tmt/package_managers/apk.py

```diff
@@ -0,0 +1,164 @@
+import re
+from typing import Optional, Union
+
+import tmt.package_managers
+import tmt.utils
+from tmt.package_managers import (
+    FileSystemPath,
+    Installable,
+    Options,
+    Package,
+    PackagePath,
+    escape_installables,
+    provides_package_manager,
+    )
+from tmt.utils import (
+    Command,
+    CommandOutput,
+    GeneralError,
+    RunError,
+    ShellScript,
+    )
+
+ReducedPackages = list[Union[Package, PackagePath]]
+
+PACKAGE_PATH = {
+    '/usr/bin/flock': 'flock'
+    }
+
+
+@provides_package_manager('apk')
+class Apk(tmt.package_managers.PackageManager):
+    probe_command = Command('apk', '--version')
+
+    install_command = Command('add')
+
+    _sudo_prefix: Command
+
+    def prepare_command(self) -> tuple[Command, Command]:
+        """ Prepare installation command for apk """
+
+        if self.guest.facts.is_superuser is False:
+            self._sudo_prefix = Command('sudo')
+
+        else:
+            self._sudo_prefix = Command()
+
+        command = Command()
+
+        command += self._sudo_prefix
+        command += Command('apk')
+
+        return (command, Command())
+
+    def path_to_package(self, path: FileSystemPath) -> Package:
+        """
+        Find a package providing given filesystem path.
+
+        This is not easily possible in Alpine. There is `apk-file` utility
+        available but it seems unrealiable. Support only a fixed set
+        of mappings until a better solution is available.
+        """
+
+        string_path = str(path)
```

Comment:
This suggests `PACKAGE_PATH` should be probably `dict[FileSystemPath, str]` than `dict[str, str]`.

Reply:
will check thx

Reply:
addressed